src/Vault.sol

```diff
@@ -209,7 +210,7 @@ contract Vault is SemiFungibleVault, ReentrancyGuard {
                 uint256 premium = entitledShares - assets;
```

Comment:
change `previewWithdraw` to 
```solidity
   function previewWithdraw(uint256 id, uint256 assets)
        public
        view
        override
        returns (uint256 entitledAmount)
    {
        entitledAmount = assets.mulDivUp(idClaimTVL[id] - epochTreasuryFee[id], idFinalTVL[id]);
    }
```

change `withdraw` to 

```solidity
function withdraw(
        uint256 id,
        uint256 assets,
        address receiver,
        address owner
    )
        external
        override
        epochHasEnded(id)
        marketExists(id)
        returns (uint256 shares)
    {
        if (receiver == address(0)) revert AddressZero();

        if (msg.sender != owner && isApprovedForAll(owner, msg.sender) == false)
            revert OwnerDidNotAuthorize(msg.sender, owner);

        uint256 entitledShares;
        _burn(owner, id, assets);

        if (epochNull[id] == false) {
            entitledShares = previewWithdraw(id, assets);
        } else {
            entitledShares = assets;
        }
        if (entitledShares > 0) {
            assert(asset.transfer(receiver, entitledShares));
        }

        emit Withdraw(msg.sender, receiver, owner, id, assets, entitledShares);

        return entitledShares;
    }

Reply:
preview withdraw with ``` - epochTreasuryFee[id] ``` faces some rounding errors that make withdraw not possible due to missing eth for some users to withdraw

Reply:
there shouldn't be any ETH missing

Reply:
separate issue, marked as resolved